crates/lint/src/sol/mod.rs

```diff
@@ -0,0 +1,286 @@
+        });
+
+        sess.emitted_diagnostics()
+    }
+}
+
+impl Linter for SolidityLinter {
+    type Language = SolcLanguage;
+    type Lint = SolLint;
+
+    fn lint(&self, input: &[PathBuf]) {
+        input.into_par_iter().for_each(|file| {
+            _ = self.lint_file(file);
+        });
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum SolLintError {
+    #[error("Unknown lint ID: {0}")]
+    InvalidId(String),
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct SolLint {
+    id: &'static str,
+    description: &'static str,
+    help: Option<&'static str>,
+    severity: Severity,
+}
+
+impl Lint for SolLint {
+    fn id(&self) -> &'static str {
+        self.id
+    }
+    fn severity(&self) -> Severity {
+        self.severity
+    }
+    fn description(&self) -> &'static str {
+        self.description
+    }
+    fn help(&self) -> Option<&'static str> {
+        self.help
+    }
+}
+
+macro_rules! declare_forge_lints {
+    ($(($pass_id:ident, $lint_id:ident, $severity:expr, $str_id:expr, $description:expr, $help:expr)),* $(,)?) => {
+        // Declare the static `Lint` metadata
+        $(
+            pub static $lint_id: SolLint = SolLint {
+                id: $str_id,
+                severity: $severity,
+                description: $description,
+                help: if $help.is_empty() { None } else { Some($help) }
+            };
+        )*
+
+        // Implement TryFrom<&str> for `SolLint`
+        impl<'a> TryFrom<&'a str> for SolLint {
+            type Error = SolLintError;
+
+            fn try_from(value: &'a str) -> Result<Self, Self::Error> {
+                match value {
+                    $(
+                        $str_id => Ok($lint_id),
+                    )*
+                    _ => Err(SolLintError::InvalidId(value.to_string())),
+                }
+            }
+        }
+
+        // Declare the structs that will implement the pass trait
+        $(
+            #[derive(Debug, Default, Clone, Copy, Eq, PartialEq)]
+            pub struct $pass_id;
+        )*
+    };
+}
+
+// Macro for defining lints and relevant metadata for the Solidity linter.
+//
+// This macro generates the [`SolLint`] enum with each lint along with utility methods and
+// corresponding structs for each lint specified.
+//
+// # Parameters
+//
+// Each lint is defined as a tuple with the following fields:
+// - `$id`: Identitifier used as the struct and enum variant created for the lint.
+// - `$severity`: The [`Severity`] of the lint (e.g. `High`, `Med`, `Low`, `Info`, `Gas`).
+// - `$description`: A short description of the lint.
+// - `$help`: Link to additional information about the lint or best practices.
+// - `$str_id`: A unique identifier used to reference a specific lint during configuration.
+declare_forge_lints!(
+    //High
+    (
+        IncorrectShift,
+        INCORRECT_SHIFT,
+        Severity::High,
+        "incorrect-shift",
+        "The order of args in a shift operation is incorrect",
```

Comment:
I am not sure if the displayed message should be declared here; in any case if it's displayed it should be lowercase since it comes after a colon `: `